AP-protocol.md

```diff
@@ -278,29 +315,86 @@ Once that the client or server has received all fragments (that is, `fragment_in
 Therefore, the packet is now a message that can be delivered.
 
 # Drone Protocol
-When a drone receives a packet, it **must** do the following:
 
-1. increase `hop_index` by 1
-2. obtain the (new `hop_index`) + 1 element of the `SourceRoutingHeader` vector `hops`, let's call it `next_hop`
-	* It **must ignore** intentionally to check `hop_index`.
+When a drone receives a packet, it **must** perform the following steps:
 
-3. if `next_hop`
-	* doesn't exist create a new packet of type Nack, precisely of type `DestinationIsDrone`. The packet must have the routing made of a vector but inverted and only contains the nodes from this drone to the sender. Send this packet as a normal packet. End here.
-	* if the `NodeId` is not a neighbor, then creates a new packet of type Nack, precisely of type `ErrorInRouting` with field the value of `NodeId` of next hop. Continue as other error.
+1. **Step 1**: Check if `hops[hop_index]` matches the drone's own `NodeId`.
+	- **If yes**, proceed to Step 2.
+	- **If no**, send a Nack with `ErrorInRouting` (including the drone's own `NodeId`) and terminate processing.
 
-4. Proceed as follows based on packet type:
+2. **Step 2**: Increment `hop_index` by **1**.
 
-### Flood Messages
-TODO  (If the packet is flood related, follow the rules in the flood section)
+3. **Step 3**: Determine if the drone is the final destination:
+	- **If `hop_index` equals the length of `hops`**, the drone is the final destination and processes the packet accordingly.
```

Comment:
Processing here means always sending an error. Because this is not specified anywhere else in the protol please add it back here.

Reply:
yes, you are right